tests/testthat/test-create_avg_best_case_worst_case_at_product_level.R

```diff
@@ -0,0 +1,50 @@
+test_that("single `NA` in `transition_risk_category` gives correct values for `avg_transition_risk_best_case` and `avg_transition_risk_worst_case`", {
+  example_data <- example_best_case_worst_case_transition_risk_profile_product_level(
+    transition_risk_category = c("low", "low", NA_character_)
+  )
+
+  out <- create_avg_best_case_worst_case_at_product_level(example_data)
+
+  expected_best_case <- 2/3
+  expected_worst_case <- 2/3
+  expect_equal(unique(out$avg_transition_risk_best_case), expected_best_case)
+  expect_equal(unique(out$avg_transition_risk_worst_case), expected_worst_case)
+})
+
+test_that("only `NA` in `transition_risk_category` gives `0` value for `avg_transition_risk_best_case` and `avg_transition_risk_worst_case`", {
+  example_data <- example_best_case_worst_case_transition_risk_profile_product_level(
```

Comment:
No big deal at all but in this case `NA` is simpler than `c(NA, NA, NA)`  and yields the same:

``` r
example <- tiltIndicator:::example_data_factory(tibble::tibble(x = 1:3))
example(y = NA_character_)
#> # A tibble: 3 × 2
#>   y         x
#>   <chr> <int>
#> 1 <NA>      1
#> 2 <NA>      2
#> 3 <NA>      3
```

<sup>Created on 2024-06-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>